File: crates/weave-graph-parse/tests/fixtures/sample.java
import java.util.List;

interface Named {
    String getName();
}

class Greeter implements Named {
    private String name;

    public String getName() {
        return this.name;
    }

    public String greet() {
        return formatName(this.getName());
    }
}

class LoudGreeter extends Greeter {
}

File: crates/weave-graph-parse/tests/fixtures/sample.dart
abstract class Greeter {
  String greet();
}

class LoudGreeter extends Greeter {
  String name;

  LoudGreeter(this.name);

  String greet() {
    return formatName(name);
  }
}

String formatName(String name) {
  return name.trim();
}
